actions/lib/action.py

```diff
@@ -22,12 +22,14 @@ def __init__(self, config):
             'aws_access_key_id': None,
             'aws_secret_access_key': None
         }
+        self.userdata = None
 
-        if config['st2_user_data']:
-            with open(config['st2_user_data'], 'r') as fp:
-                self.userdata = fp.read()
-        else:
-            self.userdata = None
+        if 'st2_user_data' in config and config['st2_user_data']:
```

Comment:
Short hand notation which should also work is `config.get('st2_user_data', None)`.